[ML] Adds function reference help text to advanced wiz

szabosteve opened the pull request:
## Summary

This PR adds helper text about the different kinds of anomaly detection functions to the advanced job wizard. The function help is in a separate file and referenced in the `descriptions.tsx`.

<img width="450" alt="Screenshot 2023-03-09 at 17 11 21" src="https://user-images.githubusercontent.com/22324794/224090073-43386e1b-4bae-41d3-a71f-43238899ebf8.png">

<img width="450" alt="Screenshot 2023-03-10 at 12 58 10" src="https://user-images.githubusercontent.com/22324794/224310405-82dc16e8-d427-45ad-9cd7-ad42914f070f.png">


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)

<!--ONMERGE {"backportTargets":["8.7"]} ONMERGE-->

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

lcawl commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+      id: 1,
+      function: 'non_zero_count, high_non_zero_count, low_non_zero_count',
+      description:
+        'Detect anomalies when the number of events in a bucket is anomalous, but it ignores cases where the bucket count is zero.',
```
Not mandatory, but you can synch the verbs here:
```suggestion
        'Detect anomalies when the number of events in a bucket is anomalous, but ignore cases where the bucket count is zero.',
```

lcawl commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+      id: 5,
+      function: 'min',
+      description:
+        'Detects anomalies in the arithmetic minimum of a value. The minimum value is calculated for each bucket.',
```
This is a little shorter:
```suggestion
        'Detect anomalies in the arithmetic minimum of a value, which is calculated for each bucket.',
```

lcawl commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+      id: 6,
+      function: 'max',
+      description:
+        'Detects anomalies in the arithmetic maximum of a value. The maximum value is calculated for each bucket.',
```
A little shorter:
```suggestion
        'Detect anomalies in the arithmetic maximum of a value, which is calculated for each bucket.',
```

lcawl commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+      id: 7,
+      function: 'median, high_median, low_median',
+      description:
+        'Detect anomalies in the statistical median of a value. The median value is calculated for each bucket.',
```
Ditto:
```suggestion
        'Detect anomalies in the statistical median of a value, which is calculated for each bucket.',
```

lcawl commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+      id: 8,
+      function: 'mean, high_mean, low_mean',
+      description:
+        'Detect anomalies in the arithmetic mean of a value. The mean value is calculated for each bucket.',
```
```suggestion
        'Detect anomalies in the arithmetic mean of a value, which is calculated for each bucket.',
```

lcawl commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+      id: 9,
+      function: 'metric',
+      description:
+        'Combines min, max, and mean functions. You can use it as a shorthand for a combined analysis. If you do not specify a function in a detector, this is the default function.',
```
IMO The "Use it as a shorthand..." sentence could probably also be dropped.
```suggestion
        'Combine min, max, and mean functions. Use it as a shorthand for a combined analysis. This is the default function.',
```

lcawl reviewed (approved):
Some minor text suggestions, but LGTM!

jgowdyelastic commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,192 @@
+    {
+      id: 0,
+      function: 'count, high_count, low_count',
+      description: 'Detect anomalies when the number of events in a bucket is anomalous.',
```
all text should be wrapped in a translation function 
e.g.

```suggestion
      description: i18n.translate('xpack.ml.newJob.wizard.pickFieldsStep.advancedDetectorModal.functionHelp.count', {
        defaultMessage: 'Detect anomalies when the number of events in a bucket is anomalous.',
      }),
```

jgowdyelastic commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/descriptions.tsx:
```diff
@@ -18,19 +20,30 @@ export const AggDescription: FC = memo(({ children }) => {
-        <>{children}</>
-      </EuiFormRow>
-    </EuiDescribedFormGroup>
+    <>
```
this wrapping fragment `<>` isn't needed

jgowdyelastic commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,274 @@
+      isOpen={isHelpOpen}
+      display="inlineBlock"
+      panelPaddingSize="none"
+      className="anomalyDetectionFunctionsHelp_popover"
+      panelClassName="anomalyDetectionFunctionsHelp_panel"
```
These classes don't actually exist, so don't need to be included here.

szabosteve commented on x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/function_help.tsx:
```diff
@@ -0,0 +1,274 @@
+
+  const helpButton = <EuiButtonIcon onClick={onHelpClick} iconType="documentation" />;
+
+  const columnsSidebar = [
```
```suggestion
  const columns = [
```

szabosteve commented:
@jgowdyelastic Thank you for your review. I addressed your suggestions via [b87a394](https://github.com/elastic/kibana/pull/153009/commits/b87a394076c7a2c244958e86bd81609e380df100).

jgowdyelastic commented:
I'm wondering about the icon being used here.
![image](https://user-images.githubusercontent.com/22172091/226857716-54b681dd-2c68-4933-bd72-29bf621b602b.png)

We now have a few different icons being used for "info"
![image](https://user-images.githubusercontent.com/22172091/226857565-76102fef-1f5c-4197-bb5f-e4a0620dae71.png)
![image](https://user-images.githubusercontent.com/22172091/226858196-21783f62-e824-4613-82e6-cd14ad831f7a.png)


and 
![image](https://user-images.githubusercontent.com/22172091/226857915-f44e2e94-4c75-4b30-b55c-55beb64857a8.png)

and 
![image](https://user-images.githubusercontent.com/22172091/226858025-2e3f5129-856c-4392-8eeb-1e05477b4b1f.png)

maybe we should go with the `help` icon for consistency.
![image](https://user-images.githubusercontent.com/22172091/226858609-93f6bdb6-6fef-4906-8a53-a2c820e43f8e.png)

szabosteve commented:
Great catch, thank you! I change the icon type to 'help' from 'documentation' in [8ca215f](https://github.com/elastic/kibana/pull/153009/commits/8ca215f7ce9bbe9c2c9ab0756a57df41654a2a7b).

jgowdyelastic reviewed (approved):
LGTM

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.7|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/153434">](https://github.com/elastic/kibana/pull/153434)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
